#### Atividade 6 - Algoritmos de busca sem informação

##### 1. Escolha duas cidades para buscar uma rota usando a implementação da busca em largura, mostre o passo a passo da busca.

In [52]:
import busca_largura as bl

In [53]:
# Escolheremos as duas cidades aleatoriamente

import random as rnd

rnd.seed(1)
(estado_inicial, estado_objetivo) = rnd.sample(bl.estados_romenia, 2)

print(f"{estado_inicial['estado']} -> {estado_objetivo['estado']}")

Oradea -> Iasi


In [54]:
no_Oradea = bl.No('Oradea', 0, None, None)

problema_Oradea_Iasi = bl.Problema(bl.estados_romenia, no_Oradea, lambda no: no.estado == 'Iasi')

In [55]:
busca_Oradea_Iasi = bl.BuscaLargura(problema_Oradea_Iasi)
busca_Oradea_Iasi.fronteira

[(Oradea, 0)]

A busca começa em Oradea, com fronteira vazia

A seguir, os passos dados até chegar no objetivo

In [56]:
busca_Oradea_Iasi.passo_busca()
busca_Oradea_Iasi.fronteira

[(Zerind, 71), (Sibiu, 151)]

In [57]:
busca_Oradea_Iasi.passo_busca()
busca_Oradea_Iasi.fronteira

[(Sibiu, 151), (Arad, 146)]

In [58]:
busca_Oradea_Iasi.passo_busca()
busca_Oradea_Iasi.fronteira

[(Arad, 146), (Fagaras, 250), (Rimnicu Vilcea, 231)]

In [59]:
busca_Oradea_Iasi.passo_busca()
busca_Oradea_Iasi.fronteira

[(Fagaras, 250), (Rimnicu Vilcea, 231), (Timisoara, 264)]

In [60]:
busca_Oradea_Iasi.passo_busca()
busca_Oradea_Iasi.fronteira

[(Rimnicu Vilcea, 231), (Timisoara, 264), (Bucharest, 461)]

In [61]:
busca_Oradea_Iasi.passo_busca()
busca_Oradea_Iasi.fronteira

[(Timisoara, 264), (Bucharest, 461), (Craiova, 377), (Pitesti, 328)]

In [62]:
busca_Oradea_Iasi.passo_busca()
busca_Oradea_Iasi.fronteira

[(Bucharest, 461), (Craiova, 377), (Pitesti, 328), (Lugoj, 375)]

In [63]:
busca_Oradea_Iasi.passo_busca()
busca_Oradea_Iasi.fronteira

[(Craiova, 377), (Pitesti, 328), (Lugoj, 375), (Giurgiu, 551), (Urziceni, 546)]

In [64]:
busca_Oradea_Iasi.passo_busca()
busca_Oradea_Iasi.fronteira

[(Pitesti, 328), (Lugoj, 375), (Giurgiu, 551), (Urziceni, 546), (Drobeta, 497)]

In [65]:
busca_Oradea_Iasi.passo_busca()
busca_Oradea_Iasi.fronteira

[(Lugoj, 375), (Giurgiu, 551), (Urziceni, 546), (Drobeta, 497)]

In [66]:
busca_Oradea_Iasi.passo_busca()
busca_Oradea_Iasi.fronteira

[(Giurgiu, 551), (Urziceni, 546), (Drobeta, 497), (Mehadia, 445)]

In [67]:
busca_Oradea_Iasi.passo_busca()
busca_Oradea_Iasi.fronteira

[(Urziceni, 546), (Drobeta, 497), (Mehadia, 445)]

In [68]:
busca_Oradea_Iasi.passo_busca()
busca_Oradea_Iasi.fronteira

[(Drobeta, 497), (Mehadia, 445), (Vaslui, 688), (Hirsova, 644)]

In [69]:
busca_Oradea_Iasi.passo_busca()
busca_Oradea_Iasi.fronteira

[(Mehadia, 445), (Vaslui, 688), (Hirsova, 644)]

In [70]:
busca_Oradea_Iasi.passo_busca()
busca_Oradea_Iasi.fronteira

[(Vaslui, 688), (Hirsova, 644)]

In [71]:
busca_Oradea_Iasi.passo_busca()
busca_Oradea_Iasi.fronteira

[(Hirsova, 644), (Iasi, 780)]

In [72]:
busca_Oradea_Iasi.passo_busca()
busca_Oradea_Iasi.fronteira

[(Iasi, 780), (Eforie, 730)]

In [73]:
busca_Oradea_Iasi.passo_busca()
busca_Oradea_Iasi.fronteira

[(Eforie, 730)]

No próximo passo, ele chegará a Iasi

In [74]:
busca_Oradea_Iasi.passo_busca()

Busca chegou ao objetivo com sucesso


In [75]:
busca_Oradea_Iasi.solucao

[(Oradea, 0),
 (Sibiu, 151),
 (Fagaras, 250),
 (Bucharest, 461),
 (Urziceni, 546),
 (Vaslui, 688),
 (Iasi, 780)]

A busca encontrou um caminho entre as duas cidades com sucesso e teve um custo final de 780